Makefile

```diff
@@ -558,6 +566,41 @@ ifdef LOCAL_BUILD
 	$(error LOCAL_BUILD must not be set for a release)
 endif
 
+release-prep: var-require-all-VERSION-CALICO_VERSION-COMMON_VERSION
+ifdef CNX
+	$(MAKE) var-require-all-CALICO_ENTERPRISE_VERSION
+	yq e ".title = \"$(CALICO_ENTERPRISE_VERSION)\" | .components |= with_entries(select(.key | test(\"^(eck-|coreos-).*\") | not)) |= with(.[]; .version = \"$(CALICO_ENTERPRISE_VERSION)\")" -i config/enterprise_versions.yml
+endif
+	yq e ".title = \"$(CALICO_VERSION)\" | .components.[].version = \"$(CALICO_VERSION)\"" -i config/calico_versions.yml
+	yq e ".title = \"$(COMMON_VERSION)\" | .components.key-cert-provisioner.version = \"$(COMMON_VERSION)\"" -i config/common_versions.yml
+	sed -i "s/\"gcr.io.*\"/\"quay.io\/\"/g" pkg/components/images.go
+	sed -i "s/\"gcr.io.*\"/\"quay.io\"/g" hack/gen-versions/main.go
+	$(MAKE) gen-versions release-prep/create-and-push-branch release-prep/create-pr release-prep/set-merge-when-ready-on-pr
+
+GIT_REMOTE?=origin
+GIT_PR_BRANCH_BASE?=$(SEMAPHORE_GIT_BRANCH)
+GIT_REPO_SLUG?=$(SEMAPHORE_GIT_REPO_SLUG)
+RELEASE_UPDATE_BRANCH?=semaphore-auto-build-updates-$(VERSION)
+GIT_PR_BRANCH_HEAD?=$(RELEASE_UPDATE_BRANCH)
+release-prep/create-and-push-branch:
+ifeq ($(shell git rev-parse --abbrev-ref HEAD),$(GIT_PR_BRANCH_HEAD))
+	$(error Current branch is pull request head, cannot set it up.)
+endif
+	-git branch -D $(GIT_PR_BRANCH_HEAD)
+	-$(GIT) push $(GIT_REMOTE) --delete $(GIT_PR_BRANCH_HEAD)
+	git checkout -b $(GIT_PR_BRANCH_HEAD)
+	$(GIT) add config/*_versions.yml hack/gen-versions/main.go pkg/components/* pkg/crds/*
+	$(GIT) commit -m "Automatic version updates for $(VERSION) release"
+	$(GIT) push $(GIT_REMOTE) $(GIT_PR_BRANCH_HEAD)
+
+release-prep/create-pr:
```

Comment:
It would be nice if we could also automatically set the labels on the PR to e.g. 'docs-not-required' and 'release-note-not-required', since these docs and release note required are always defaulted to true and in this case they never are. This will reduce the work required to filter the release notes after the release is done.

Reply:
i'm not sure the slash commands that marvin allows so i'll ask around to see if this will be a quick update; otherwise, it'll likely be another ticket to get that going

Reply:
created [DE-2157](https://tigera.atlassian.net/browse/DE-2157) to address this